src/backend/catalog/aclchk.c

```diff
@@ -5005,14 +5006,30 @@ RemoveRoleFromInitPriv(Oid roleid, Oid classid, Oid objid, int32 objsubid)
 	 * Generate new ACL.  Grantor of rights is always the same as the owner.
 	 */
 	if (old_acl != NULL)
+	{
+		Oid sysadminOid;
+
+		/*
+		 * For TSQL system roles, grantor will be sysadmin instead of owner.
+		 */
+		if (bbf_get_sysadmin_oid_hook &&
+			classid == DatabaseRelationId &&
+			is_member_of_role(GetUserId(), sysadminOid = (*bbf_get_sysadmin_oid_hook)()))
+		{
+			grantorId = sysadminOid;
+		}
```

Comment:
1. Will Grantor always be sysadmin in case of Babelfish? At the time of inserting the init_privs, how are we deciding the grantor? If we see possibility that grantor being diff. from sysadmin then we should utilise `select_best_grantor()`
2. Will it by triggered for other physical databases? Ideally it shouldn't. If it is then, I would suggest add check for babelfish physical db oid comparison if possible.
3. We should check whether it is babelfish role or not as well.

Reply:
Discussed it offline and added check for babelfish database.